TransformFusion/Logic/vtkMRMLTransformFusionNode.cxx

```diff
@@ -52,91 +66,309 @@ vtkMRMLTransformFusionNode::~vtkMRMLTransformFusionNode()
 {
   std::cerr << "Reading TransformFusion parameter node" << std::endl;
   Superclass::ReadXMLAttributes(atts);
 
   const char* attName;
   const char* attValue;
-  while (*atts != NULL){
-    attName = *(atts++);
-    attValue = *(atts++);
+  while ( *atts != NULL )
+  {
+    attName = *( atts++ );
+    attValue = *( atts++ );
     
-    if (!strcmp(attName,"UpdatesPerSecond")){
+    if ( strcmp( attName, "UpdatesPerSecond" ) == 0 )
+    {
       std::stringstream ss;
       ss << attValue;
       ss >> this->UpdatesPerSecond;
       continue;
-    }    
+    }
+    else if ( strcmp( attName, "UpdateMode" ) == 0 )
+    {
+      int modeAsInt = GetUpdateModeFromString( attName );
+      if ( modeAsInt >= 0 && modeAsInt < UPDATE_MODE_LAST)
+      {
+        this->UpdateMode = modeAsInt;
+      }
+      else
+      {
+        // default
+        this->UpdateMode = UPDATE_MODE_MANUAL;
+      }
+    }
+    else if ( strcmp( attName, "FusionMode" ) == 0 )
+    {
+      int modeAsInt = GetFusionModeFromString( attName );
+      if ( modeAsInt >= 0 && modeAsInt < FUSION_MODE_LAST)
+      {
+        this->FusionMode = modeAsInt;
+      }
+      else
+      {
+        // default
+        this->FusionMode = FUSION_MODE_QUATERNION_AVERAGE;
+      }
+    }
   }
 
-  this->WriteXML(std::cout,1);
+  this->Modified();
 }
 
 //----------------------------------------------------------------------------
-void vtkMRMLTransformFusionNode::WriteXML(ostream& of, int nIndent)
+void vtkMRMLTransformFusionNode::WriteXML( ostream& of, int nIndent )
 {
-  Superclass::WriteXML(of, nIndent);
-  vtkIndent indent(nIndent);
-  
-  of << indent << " UpdatesPerSecond=\""<< this->UpdatesPerSecond << "\"";    
+  Superclass::WriteXML( of, nIndent );
+  vtkIndent indent( nIndent );
+  of << indent << " UpdatesPerSecond=\"" << this->UpdatesPerSecond << "\"";
+  of << indent << " UpdateMode=\"" << GetUpdateModeAsString( this->UpdateMode ) << "\"";
+  of << indent << " FusionMode=\"" << GetFusionModeAsString( this->FusionMode ) << "\"";
 }
 
 //----------------------------------------------------------------------------
-void vtkMRMLTransformFusionNode::Copy(vtkMRMLNode *anode)
+void vtkMRMLTransformFusionNode::PrintSelf( ostream& os, vtkIndent indent )
 {
-  Superclass::Copy(anode);
-  vtkMRMLTransformFusionNode *node = vtkMRMLTransformFusionNode::SafeDownCast(anode);
+  Superclass::PrintSelf( os, indent );
+  os << indent << " UpdatesPerSecond = " << this->UpdatesPerSecond << "\n";
+  os << indent << " UpdateMode = " << GetUpdateModeAsString( this->UpdateMode ) << "\n";
+  os << indent << " FusionMode = " << GetFusionModeAsString( this->FusionMode ) << "\n";
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLTransformFusionNode::Copy( vtkMRMLNode *anode )
+{
+  Superclass::Copy( anode );
+  vtkMRMLTransformFusionNode *node = vtkMRMLTransformFusionNode::SafeDownCast( anode );
   this->DisableModifiedEventOn();
```

Comment:
Search for StartModify anywhere in Slicer or SlicerIGT, but it's really simple:
  int wasModifying = node->StartModify();
  ...make several modifications to node
  node->EndModify( wasModifying );